aries_cloudagent/resolver/base.py

```diff
@@ -95,8 +120,22 @@ async def resolve(self, profile: Profile, did: Union[str, DID]) -> DIDDocument:
                 vm_allow_type_list,
             },
         )
-        return result
+        return Resolution(result, resolver_metadata)
 
     @abstractmethod
     async def _resolve(self, profile: Profile, did: str) -> dict:
         """Resolve a DID using this resolver."""
+
+    async def _retrieve_resolver_metadata(self, method, previous_time):
+
+        time_now = datetime.utcnow()
+        duration = int((time_now - previous_time).total_seconds() * 1000)
+        resolver_metadata = {
+            "type": self.type,
+            "driverId": f"did:{method}",
+            "driver": self.driver,
```

Comment:
We haven't used the "Driver" terminology to this point so I'm hesitant to start now. I could see having a "resolver" item with the fully qualified class path to the resolver used, perhaps?